Chat.Try/Pages/DisplayMessages.razor

```diff
@@ -1,55 +1,122 @@
 @inject NavigationManager NavigationManager
 @inject IJSRuntime JsRuntime
 @implements IAsyncDisposable
 
-
-@if (DisplayMessageList.Any())
+<div id="scrollbox" class="col-6">
+@foreach (var message in DisplayMessageList)
 {
-    <div id="scrollbox">
-    @foreach (var message in DisplayMessageList)
+    @if (message.UserId == UserId)
     {
-        @if (message.UserId == UserId)
-        {
-            <div class="sent">
-                <div class="user">@message.UserId</div>
-                <div class="msg">@message.Message</div>
-            </div>
-        }
-        else
-        {
-            <div class="received">
-                <div class="user">@message.UserId</div>
-                <div class="msg">@message.Message</div>
-            </div>
-        }
-
+        <div class="bg-secondary">
+            <div class="user">@message.UserId</div>
+            <div class="msg">@message.Message</div>
+            <div class="small">@message.CreatedOn.ToString("F")</div>
+        </div>
+    }
+    else
+    {
+        <div class=" bg-success">
+            <div class="user">@message.UserId</div>
+            <div class="msg">@message.Message</div>
+            <div class="small">@message.CreatedOn.ToString("F")</div>
+        </div>
     }
-        <hr />
-        @*<textarea class="input-lg" placeholder="enter your comment" @bind="@_newMessage"></textarea>
-        <button class="btn btn-default" @onclick="@(() => SendAsync(_newMessage))">Send</button>*@
-    </div>
+
 }
+    <hr />
+    <textarea class="input-lg" placeholder="enter your comment" @bind="@NewMessage"></textarea>
+    <button class="btn btn-default" @onclick="@(() => SendAsync(NewMessage))">Send</button>
+</div>
 
 @code {
+    [Parameter]
+    public Conversations Conversation { get; set; } 
+    [Parameter]
+    public string UserId { get; set; }
+    [Parameter]
+    public string OtherUserId { get; set; }
+
+    private HubConnection HubConnection;
     private List<DisplayMessage> DisplayMessageList = new();
-    private string UserId { get; set; }
+    private string NewMessage { get; set; }
+
+    public bool IsConnected => HubConnection.State == HubConnectionState.Connected;
 
-    public async Task GetMessages(Conversations conversation)
+    protected override async Task OnInitializedAsync()
     {
-        var messages = conversation.ConversationUsers.SelectMany(x => x.UserMessages)
+        var timer = new System.Threading.Timer((_) =>
+        {
+            InvokeAsync(async () =>
+            {
+                // Update the UI
+                var tempMessages = _chatAccessor.GetUserMessages(Conversation.Id).Select(x => new DisplayMessage
+                {
+                    UserId = x.ConversationUser.UserId,
+                    Message = x.Message,
+                    CreatedOn = x.CreatedOn
+                });
+                DisplayMessageList = tempMessages.OrderBy(x => x.CreatedOn).ToList();
```

Comment:
this could be better...by only refreshing if the number of messages have changed